numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1144,7 +1157,17 @@ PyArray_Choose(PyArrayObject *ip, PyObject *op, PyArrayObject *out,
     PyDataMem_FREE(mps);
     if (out != NULL && out != obj) {
         Py_INCREF(out);
-        PyArray_ResolveWritebackIfCopy(obj);
+        int success;
+        if (is_newarr) {
+            success = PyArray_CopyAnyInto(out, obj);
```

Comment:
In theory, you could set the base-object to your allocation above and set to use `WritebackIfCopy` even with your custom allocation.  But, TBH, I think a direct copy is actually clearer (frankly, I am not sure if we couldn't use it in half the places `ResolveWritebackIfCopy` is used ;)).

Reply:
I see--good to know!